src/security.c

```
@@ -1,1012 +1,616 @@
+        return -1;
     }
 
-    /* Copy the public key */
-    memcpy(public_key, stored_public, PUBLIC_KEY_LENGTH);
+    /* Use domain separation with peer_id to derive a pairwise secret suitable for GROUP_JOIN encryption */
+    uint8_t context[24];
+    size_t off = 0;
+    memcpy(context + off, &peer_id, sizeof(peer_id));      off += sizeof(peer_id);
+    memcpy(context + off, "MERIDIAN_GROUP_JOIN_V1", sizeof("MERIDIAN_GROUP_JOIN_V1") - 1);
+
+    uint8_t derived[SHARED_SECRET_LENGTH];
```

Clang (14.0.6)

[warning] variable 'derived' is not initialized

(cppcoreguidelines-init-variables)